packages/shared/src/effect.ts

```diff
@@ -73,17 +107,16 @@ export const fetchEffJson = <Schema>(
         catch: (error) => new FetchError({ error, input }),
       }),
     ),
-    Effect.andThen(({ ok, json, status }) =>
-      ok
-        ? Effect.succeed(json)
-        : Effect.fail(
-            new FetchError({
-              error: `Request to ${requestUrl} failed with status ${status}`,
-              data: json as Json,
-              input,
-            }),
-          ),
+    Effect.filterOrFail(
+      (_) => _.ok,
+      ({ json, status }) =>
+        new FetchError({
+          error: `Request to ${requestUrl} failed with status ${status}`,
+          data: json as Json,
+          input,
+        }),
```

Comment:
imo `flatMap(() => cond ? succeed(val) : fail(error))` is a bit smelly. It works just find and is really simple, it's even idiomatic since it's a defacto `filterMap`, but it just reads meh. `filterOrFail` immediately exposes the _semantics_ expressed by the tertiary.

In the end, this is just stylistic